### ISSUE TITLE
🐛 Internal error when renaming a file being watched

### ISSUE THREAD

**Issue by EduardoLopes** [rome/tools#1068]
## Environment

```
$ rome rage --summary
```
<details>
<summary>
Summary
</summary>

```
Environment Variables
  Object {
    COLORTERM: "truecolor"
    LANG: "en_US.UTF-8"   
    ROME_CACHE: "1"       
    TERM_PROGRAM: "vscode"
  }

User Config
  unset

Rome Version
  10.0.4-beta

Node Version
  12.16.3

Platform
  win32 x64 10.0.18362

Terminal Features
  Object {
    background: "unknown"
    colorDepth: 24
    columns: 159
    cursor: true
    hyperlinks: true
    isTTY: true
    unicode: false
  }

Client Flags
  Object {
    clientName: "cli"
    cwd: "C:\\Users\\Eduardo\\Documents\\prog\\argila\\argila"
    realCwd: "C:\\Users\\Eduardo\\Documents\\prog\\argila\\argila"
    silent: false
  }

Server Status
  Object {
    projects: Array [
      Object {id: 0}
      Object {id: 1}
    ]
    server: Object {
      heapTotal: 27_942_912
      pid: 2_332
      uptime: 0.7726471
    }
    workers: Array [
      Object {
        astCacheSize: 0
        heapTotal: 27_942_912
        ownedBytes: 0
        ownedFileCount: 0
        pid: 2_332
        uptime: 0.7725237
      }
    ]
  }

Done in 1.87s.
```
</details>

## Steps To Reproduce

1. run `rome check --watch`
2. rename a file being watched

I got the `lint/js/useDefaultExportBasename` diagnostic, once i renamed the file, there was the internal fatal error.
I was renaming `src/systems/input/Input.ts` to `src/systems/input/InputSystem.ts`

<details>
<summary>
Log
</summary>

```
Linting: src/systems/input/InputSystem.ts                                                                                         elapsed 0.06s eta 0.06s 1/2
 server project-rome/@internal/core/server/project/ProjectManager.ts:829:9 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Expected existing project for C:\Users\Eduardo\AppData\Local\Temp\rome-10.0.4-beta\argila\src\systems\input\Input.ts_ only have / ,
    C:\Users\Eduardo\AppData\Local\Temp\rome-remote, C:\Users\Eduardo\Documents\prog\argila\argila

  1. ___R$project$rome$$internal$core$server$project$ProjectManager_ts$default.assertProjectExisting (
     project-rome/@internal/core/server/project/ProjectManager.ts:829:9)
  2. ___R$project$rome$$internal$core$server$project$ProjectManager_ts$default.getUid (project-rome/@internal/core/server/project/ProjectManager.ts:312:23)
  3. ___R$project$rome$$internal$core$server$Cache_ts$default.getCacheFilename (project-rome/@internal/core/server/Cache.ts:209:41)
  4. ___R$project$rome$$internal$core$server$Cache_ts$default.handleDeleted (project-rome/@internal/core/server/Cache.ts:216:29)
  5. handleDeleted (project-rome/@internal/core/server/Cache.ts:109:28)
  6. ___R$project$rome$$internal$events$EventQueue_ts$default.flush (project-rome/@internal/events/EventQueue.ts:39:9)
  7. runNextTicks (internal/process/task_queues.js:62:4)
  8. listOnTimeout (internal/timers.js:518:8)
  9. process.processTimers (internal/timers.js:492:6)

  ! Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.

(node:14680) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 16)
 server project-rome/@internal/events/Bridge.ts:351:20 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Connection died

  1. ___R$project$rome$$internal$core$common$bridges$WorkerBridge_ts$default.end (project-rome/@internal/events/Bridge.ts:351:20)
  2. ___R$project$rome$$internal$core$server$WorkerManager_ts$default.end (project-rome/@internal/core/server/WorkerManager.ts:106:11)
  3. ___R$project$rome$$internal$core$server$Server_ts$default.onFatalError (project-rome/@internal/core/server/Server.ts:335:21)
  4. onFatalError (project-rome/@internal/core/server/Server.ts:459:8)
  5. <anonymous> (project-rome/@internal/core/common/utils/setupGlobalErrorHandlers.ts:26:3)
  6. processTicksAndRejections (internal/process/task_queues.js:97:4)
  7. runNextTicks (internal/process/task_queues.js:66:2)
  8. listOnTimeout (internal/timers.js:518:8)
  9. process.processTimers (internal/timers.js:492:6)

  ! Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.

 project-rome/@internal/events/Bridge.ts:351:20 internalError/request ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Connection died

  1. ___R$project$rome$$internal$core$common$bridges$WorkerBridge_ts$default.end (project-rome/@internal/events/Bridge.ts:351:20)
  2. ___R$project$rome$$internal$core$server$WorkerManager_ts$default.end (project-rome/@internal/core/server/WorkerManager.ts:106:11)
  3. ___R$project$rome$$internal$core$server$Server_ts$default.onFatalError (project-rome/@internal/core/server/Server.ts:335:21)
  4. onFatalError (project-rome/@internal/core/server/Server.ts:459:8)
  5. <anonymous> (project-rome/@internal/core/common/utils/setupGlobalErrorHandlers.ts:26:3)
  6. processTicksAndRejections (internal/process/task_queues.js:97:4)
  7. runNextTicks (internal/process/task_queues.js:66:2)
  8. listOnTimeout (internal/timers.js:518:8)
  9. process.processTimers (internal/timers.js:492:6)

  i Error occurred while requesting lint for argila/src/systems/input/InputSystem.ts

(node:14680) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 17)
(node:14680) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 18)
 worker 3 project-rome/@internal/core/server/WorkerManager.ts:323:5 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Process unexpectedly exit

  1. ChildProcess.<anonymous> (project-rome/@internal/core/server/WorkerManager.ts:323:5)
  2. Object.onceWrapper (events.js:417:25)
  3. ChildProcess.emit (events.js:310:19)
  4. Process.onexit (internal/child_process.js:275:11)

  ! Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.

 worker 2 project-rome/@internal/core/server/WorkerManager.ts:323:5 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Process unexpectedly exit

  1. ChildProcess.<anonymous> (project-rome/@internal/core/server/WorkerManager.ts:323:5)
  2. Object.onceWrapper (events.js:417:25)
  3. ChildProcess.emit (events.js:310:19)
  4. Process.onexit (internal/child_process.js:275:11)

  ! Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.

 worker 1 project-rome/@internal/core/server/WorkerManager.ts:323:5 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Process unexpectedly exit

  1. ChildProcess.<anonymous> (project-rome/@internal/core/server/WorkerManager.ts:323:5)
  2. Object.onceWrapper (events.js:417:25)
  3. ChildProcess.emit (events.js:310:19)
  4. Process.onexit (internal/child_process.js:275:11)

  ! Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.

 worker 4 project-rome/@internal/core/server/WorkerManager.ts:323:5 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  × Process unexpectedly exit

  1. ChildProcess.<anonymous> (project-rome/@internal/core/server/WorkerManager.ts:323:5)
  2. Object.onceWrapper (events.js:417:25)
  3. ChildProcess.emit (events.js:310:19)
  4. Process.onexit (internal/child_process.js:275:11)

  ! Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.
```
</details>

**Comment by sebmck** [comment]
There's some funky race conditions happening in general when files are modified :(

I think to reliably fix this we need to have more reliable locks, likely global, when performing file change operations/events.

**Comment by jamiebuilds** [comment]
Hi! Thank you for opening this issue

We recently made the decision to [migrate the Rome codebase to Rust](https://rome.tools/blog/2021/09/21/rome-will-be-rewritten-in-rust) and making some significant changes to its architecture. We've also decided not to release the previous version written in TypeScript.

We're going through and closing pull requests and issues that were specific to the TypeScript codebase, including this one. Thank you for taking the time to open it.

Thanks again!
